internal/stacks/stackruntime/internal/stackeval/main.go

```diff
@@ -94,6 +94,9 @@ type mainPlanning struct {
 	// This is a utility for unit tests that want to encourage stable output
 	// to assert against. Not for real use.
 	forcePlanTimestamp *time.Time
+
+	// This is the plan timestamp that will be used for the plan if forcePlanTimestamp is not set.
+	planTimestamp *time.Time
```

Comment:
Do we still need this combination of forcePlanTimestamp and planTimestamp? Note, there's another definition of ForcePlanTimestamp within the `opts` as well, giving us three possible sources of truth about what the plan timestamp actually is.

There's also a kind of unpredictability here with the way the `PlanTimestamp` function lazily computes the timestamp as it is requested. This means that the `plantimestamp` can sometimes be during or even after the actual planning has happened.

I wonder if instead we should get rid of any concept of forcing the timestamp, and just say that this struct expects/requires the external caller (whether that's a test or [NewForPlanning](https://github.com/hashicorp/terraform/blob/af498dd2556a64c166198e516b811630e816642e/internal/stacks/stackruntime/internal/stackeval/main.go#L122)) to specify a specific time that is constant to "this is when the plan started". Essentially, the plan timestamp is always "forced", it's just set externally to the actual planning operation.

What do you think? FWIW, I'd definitely get rid of storing it both here and in the `opts` where both are accessible and could potentially be different. I'd probably stick a single `planTimestamp` in the opts, make it non-nil, and just have it be assigned when the `mainPlanning` is created, and then have the function just look at the opts directly like it does when applying.

Reply:
Yeah I can see your point, I'll just store a timestamp in here and make the caller put in the opts.ForcePlanTimestamp or the current timing 👍